scripts/resolve_i64/main.go

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	//"github.com/uber/zanzibar/codegen"
+	"go.uber.org/thriftrw/compile"
+	"html/template"
+	"bytes"
+	"io/ioutil"
+	"github.com/uber/zanzibar/runtime"
+	"runtime"
+	"path/filepath"
+	"strings"
+	"github.com/uber/zanzibar/codegen"
+	"github.com/pkg/errors"
+	"sort"
+)
+
+const (
+	templateFileName = "i64.tmpl"
+	outFileName = "/types_i64.go"
+)
+
+type naivePackageNameResolver struct {
+}
+
+func (r *naivePackageNameResolver) TypePackageName(
+	thriftFile string,
+) (string, error) {
+	if thriftFile[0] == '.' {
+		return "", errors.Errorf("Naive does not support relative imports")
+	}
+
+	return "", nil
+}
+
+// Meta is the struct container for i64 related meta data and package name
+type Meta struct {
+	PackageName string
+	Types []I64Struct
+}
+
+// I64Structs is the struct container for array if I64Struct
+type I64Structs []I64Struct
+
+// I64Struct is the struct container for i64 related meta data
+type I64Struct struct {
+	IsLong bool
+	IsTimestamp bool
+	TypedefType string
+}
+
+func (l I64Structs) Len() int           { return len(l) }
+func (l I64Structs) Less(i, j int) bool {
+	return l[i].IsTimestamp
+}
+func (l I64Structs) Swap(i, j int)      { l[i], l[j] = l[j], l[i] }
+
+func getDirName() string {
+	_, file, _, _ := runtime.Caller(0)
+	return zanzibar.GetDirnameFromRuntimeCaller(file)
+}
+
+func main() {
+	thriftFile := os.Args[1]
+	module, err := compile.Compile(thriftFile)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to parse thrift file: %s", thriftFile))
+	}
+	meta := &Meta{}
+
+	s := strings.TrimSuffix(thriftFile, ".thrift")
+	s = strings.Replace(s, "/idl/", "/build/gen-code/", 1)
```

Comment:
we can leave a TODO here. `/build/gen-code/` is hardcoded and we might want to customize that; maybe via `os.Args[2]`